db/range_del_aggregator.cc

```diff
@@ -138,159 +139,80 @@ class CollapsedRangeDelMap : public RangeDelMap {
-      // accordingly so we can use common code paths later.
-      int new_to_old_end_cmp;
-      if (new_range_dels_iter_end == nullptr && rep_iter_end == nullptr) {
-        new_to_old_end_cmp = 0;
-      } else if (new_range_dels_iter_end == nullptr) {
-        new_to_old_end_cmp = 1;
-      } else if (rep_iter_end == nullptr) {
-        new_to_old_end_cmp = -1;
-      } else {
-        new_to_old_end_cmp =
-            ucmp_->Compare(*new_range_dels_iter_end, *rep_iter_end);
-      }
-
-      if (new_to_old_start_cmp < 0) {
-        // the existing one's left endpoint comes after, so raise/delete it if
-        // it's covered.
-        if (rep_iter->second.seq_ < new_range_dels_iter->seq_) {
-          untermed_seq = rep_iter->second.seq_;
-          if (rep_iter != rep_.begin() &&
-              std::prev(rep_iter)->second.seq_ == new_range_dels_iter->seq_) {
-            rep_iter = rep_.erase(rep_iter);
-            --rep_iter;
-          } else {
-            rep_iter->second.seq_ = new_range_dels_iter->seq_;
-          }
-        }
-      } else if (new_to_old_start_cmp > 0) {
-        if (untermed_seq != kMaxSequenceNumber ||
-            rep_iter->second.seq_ < new_range_dels_iter->seq_) {
-          auto seq = rep_iter->second.seq_;
-          // need to adjust this element if not intended to span beyond the new
-          // element (i.e., was_rep_iter_raised == true), or if it
-          // can be raised
-          rep_iter = rep_.emplace(
-              new_range_dels_iter->start_key_,
-              RangeTombstone(
-                  Slice(), Slice(),
-                  std::max(
-                      untermed_seq == kMaxSequenceNumber ? 0 : untermed_seq,
-                      new_range_dels_iter->seq_)));
-          untermed_seq = seq;
-        }
-      } else {
-        // their left endpoints coincide, so raise the existing one if needed
-        if (rep_iter->second.seq_ < new_range_dels_iter->seq_) {
-          untermed_seq = rep_iter->second.seq_;
-          rep_iter->second.seq_ = new_range_dels_iter->seq_;
+    // Look all the existing transition points that overlap the new tombstone.
+    while (it != rep_.end() && ucmp_->Compare(it->first, t.end_key_) < 0) {
+      if (t.seq_ > it->second) {
+        // The new tombstone covers this existing tombstone transition point.
+        // (Remember, every entry in this map is the end of one tombstone and
+        // the start of the next.)
+
+        // Remember the covered tombstone's seqno. We might need to return to
+        // it if the existing tombstone extends to the right of
+        end_seq = it->second;
+
+        if (t.seq_ == prev_seq()) {
+          // The last transition point already transitioned to the new
+          // tombstone, so we can erase this transition point. It's redundant.
+          // (It wouldn't be incorrect to leave it in, but it unnecessarily
+          // increases the size of the map.)
+          it = rep_.erase(it);
+          it--;
```

Comment:
You're decrementing `it` here, only to increment it on line 199. Perhaps just `continue` instead.

Reply:
Done.